### PR TITLE
docs: Small dev notes for using exchange from source

### PR DESCRIPTION
**Pull request by michaelneale**

### CODE REVIEW

**Review by codefromthecrypt — APPROVED** [review]
thanks so much, I was wondering how to do this!

**Review by baxen — APPROVED** [review]
LGTM!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -30,7 +29,15 @@ just test
 
 will run a fresh goose session (can use the usual goose commands with `uv run` prefixed)
 
+## Running ai-exchange from source
 
+goose depends heavily on the https://github.com/square/exchange project, you can clone that and then work on both by running:
```
Comment (lilydelalande) [inline CONTRIBUTING.md]: nit:
```suggestion
goose depends heavily on the https://github.com/square/exchange project, you can clone that repo and then work on both by running: 
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -30,7 +29,15 @@ just test
 
 will run a fresh goose session (can use the usual goose commands with `uv run` prefixed)
 
+## Running ai-exchange from source
 
+goose depends heavily on the https://github.com/square/exchange project, you can clone that and then work on both by running: 
+
+```sh
+uv add --editable ../exchange
+```
```
Comment (lilydelalande) [inline CONTRIBUTING.md]:
```suggestion
```sh
uv add --editable <path/to/cloned/exchange>
```
```

just to be explicit

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -30,7 +29,15 @@ just test
...
+
+then when you run goose with `uv run goose` it will be running it all from source.
```
Comment (lilydelalande) [inline CONTRIBUTING.md:39]: nit:

> and changes in `exchange` will be reflected automatically

**Review by lily-de — APPROVED** [review]

**Pull request merged**